libopenage/presenter/presenter.cpp

```diff
@@ -313,11 +325,18 @@ void Presenter::render() {
 	this->terrain_renderer->update();
 	this->world_renderer->update();
 	this->hud_renderer->update();
+
 	this->gui->render();
+	this->renderer->render(this->render_passes[this->index_gui_stencil_pass]);
 
-	for (auto &pass : this->render_passes) {
-		this->renderer->render(pass);
+	for (size_t i = 0; i < this->index_gui_stencil_pass; ++i) {
+		this->renderer->render(this->render_passes[i]);
 	}
+
+	this->gui->render();
+	this->renderer->render(this->render_passes[this->index_gui_render_pass]);
+
+	this->renderer->render(this->render_passes[this->index_final_render_pass]);
```

Comment:
The GUI is now rendered twice, which is not what we want if the goal is performance ;)

Also, code like this needs comment because otherwise it becomes very confusing to understand why the order of render passes is structured this way.